pkg/prometheus/statefulset.go

```diff
@@ -127,6 +129,22 @@ func MakeConfigSecret(p monitoringv1.PrometheusInterface, config operator.Config
 	}
 }
 
+func MakeCompressedSecretForPrometheus(p monitoringv1.PrometheusInterface, config operator.Config, data []byte) (*v1.Secret, error) {
```

Comment:
(suggestion) I would 
* Merge `MakeConfigSecret` and `MakeConfigurationSecret` so the resulting function includes the compression of the raw config.
* get rid of `MakeEmptyConfigurationSecret`  as it should be equivalent to calling `MakeConfigurationSecret(p, c.config, nil)` 
```suggestion
func MakeConfigurationSecret(p monitoringv1.PrometheusInterface, config operator.Config, data []byte) (*v1.Secret, error) {
```